### PR TITLE
193 add contour line and contour angles for profiles

### LINKED ISSUE #193: Add contour line and contour angles for profiles
**Summary**

To use von-Karmans theory of rolling for pyroll, profiles need to "know" about their contour. 
Therefore, a 'contour_line' as well as `contour_angles`  hook should be included to the profile.

### PR DESCRIPTION
**Pull request by ChRen95**
closes #193

### CODE REVIEW

**Comment by axtimhaus** [comment]
Is logically not correct to define them on plain `Profile`, since this has no contact toanything. Move them to `DeformationUnit.Profile`

**Comment by ChRen95** [comment]
Reworked calculation and moved to deformation unit.

File: pyroll/core/roll_pass/hookimpls/deformation_unit.py [hunk 1]
```diff
@@ -84,3 +88,35 @@ def zener_holomon_parameter(self: DeformationUnit):
...
+        xdiff = x[2:] - x[:-2]
+        ydiff = y[2:] - y[:-2]
+        angles = np.arctan2(ydiff, xdiff)
+
+        return angles
+
+    upper_groove_cl_angles = calculate_angles(self.contact_contour_lines[0])
+    lower_groove_cl_angles = calculate_angles(self.contact_contour_lines[1])
+
+    return [upper_groove_cl_angles, lower_groove_cl_angles]
```
Comment (axtimhaus) [inline pyroll/core/roll_pass/hookimpls/deformation_unit.py:122]:
```suggestion
    return [calculate_angles(line) for line in self.contact_contour_lines]
```

to support 3-roll passes

**Review by axtimhaus — APPROVED** [review]

File: pyroll/core/roll_pass/hookimpls/deformation_unit.py (hunk 1 above)
Comment (axtimhaus) [reply pyroll/core/roll_pass/hookimpls/deformation_unit.py:122]: I thought you will notice this change without explicitly mark it as request changes?

**Pull request merged**